src/Hadamard.jl

```diff
@@ -368,6 +368,21 @@ function hadamard(n::Integer)
     return H
 end
 
+"""
+    walsh(n)
+
+Return a Walsh matrix of order `n`. by bit-reversal permutation of a Hadamard matrix.
+"""
+function walsh(n::Integer)
+    m = Int(round(log2(n))) + 1 # number of bits to represent the index
+    e = sizeof(Int)*8 - m       # number of extra bits
+    i = 0:n-1                   # Hadamard index
+    j = bitreverse.(i) .>> e    # bit reversing of the binary index
+    j = j .⊻ (j .>> 1)          # binary sequency index
+
+    return hadamard(n)[j.+1,:]
```

Comment:
This eliminates a couple of temporary arrays:
```suggestion
    j = [ let b = bitreverse(i) >> e # bit-reverse the binary index (trailing m bits)
              j = b ⊻ (b >> 1) # binary sequency index
              j + 1 # 1-based index
          end
          for i in 0:n-1 ] 

    return hadamard(n)[j, :]
```